documentation/metrics.md

```diff
@@ -0,0 +1,386 @@
+- **Type:** Counter
+- **Description:** Total number of HTTP requests
+- **Labels:**
+  - `method`: HTTP method (GET, POST, etc.)
+  - `endpoint`: Normalized endpoint path (e.g., `/within`, `/intersect`)
+  - `status_code`: HTTP response status code
+  - `topic`: Topic used in the request. When a request includes multiple topics (e.g., `["kreis", "land"]`), a separate metric is recorded for each topic
+  - `app`: Application name (always "geospatialanalyzer")
+- **Note:** For requests with multiple topics, each topic generates a separate counter increment. For example, a request with `topics: ["kreis", "land"]` will increment both `topic="kreis"` and `topic="land"` counters.
+
+#### `http_request_duration_seconds`
+- **Type:** Histogram
+- **Description:** Duration of HTTP requests in seconds
+- **Labels:**
+  - `method`: HTTP method
+  - `endpoint`: Normalized endpoint path
+  - `app`: Application name
+- **Buckets:** 0.01, 0.05, 0.1, 0.5, 1, 2, 5, 10, 30 seconds
+
+#### `http_request_size_bytes`
+- **Type:** Histogram
+- **Description:** Size of HTTP request payloads in bytes
+- **Labels:**
+  - `method`: HTTP method
+  - `endpoint`: Normalized endpoint path
+  - `app`: Application name
+- **Buckets:** 100, 1000, 10000, 100000, 1000000, 10000000 bytes
+
+#### `http_response_size_bytes`
+- **Type:** Histogram
+- **Description:** Size of HTTP response payloads in bytes
+- **Labels:**
+  - `method`: HTTP method
+  - `endpoint`: Normalized endpoint path
+  - `app`: Application name
+- **Buckets:** 100, 1000, 10000, 100000, 1000000, 10000000 bytes
+
+#### `http_active_connections`
+- **Type:** Gauge
+- **Description:** Current number of active HTTP connections
+- **Labels:**
+  - `app`: Application name
+
+### Database Metrics
+
+#### `db_query_duration_seconds`
+- **Type:** Histogram
+- **Description:** Duration of database queries in seconds
+- **Labels:**
+  - `query_type`: Type of SQL query (SELECT, INSERT, UPDATE, DELETE, etc.)
+  - `endpoint`: The API endpoint that initiated the query (e.g., `/intersect`, `/within`, `/nearestNeighbour`, `/valuesAtPoint`)
+  - `app`: Application name
+- **Buckets:** 0.001, 0.005, 0.01, 0.05, 0.1, 0.5, 1, 2, 5, 10 seconds
+
+#### `db_queries_total`
+- **Type:** Counter
+- **Description:** Total number of database queries executed
+- **Labels:**
+  - `query_type`: Type of SQL query
+  - `status`: Query execution status (success or error)
+  - `endpoint`: The API endpoint that initiated the query
+  - `app`: Application name
+
+### Application Metrics
+
+#### `query_results_returned`
+- **Type:** Histogram
+- **Description:** Number of results returned per query
+- **Labels:**
+  - `endpoint`: API endpoint path
+  - `topic`: Topic queried (e.g., "kreis", "land", "parcels")
+  - `app`: Application name
+- **Buckets:** 0, 1, 5, 10, 25, 50, 100, 250, 500, 1000, 5000 results
+
+### System Metrics
+
+The application also exposes standard Node.js metrics with the `geospatialanalyzer_` prefix:
```

Comment:
Just because I'm wondering: why are we using a prefix here but not on the other metrics (where we add an app-label)?